src/bindings/python/src/pyopenvino/graph/model.cpp

```diff
@@ -327,19 +327,44 @@ void regclass_graph_Model(py::module m) {
     model.def(
         "reshape",
         [](ov::Model& self, const py::list& partial_shape) {
-            ov::PartialShape new_shape(Common::partial_shape_from_list(partial_shape));
-            py::gil_scoped_release release;
-            self.reshape(new_shape);
+            if (py::isinstance<py::list>(partial_shape[0])) {
```

Comment:
what if user put empty list corresponding to scalar? Example: `[]`

Reply:
Adding that check and test as well, it's taking a lot of time to build again. will commit once it's done

Reply:
Please cover this in this PR

Reply:
Yes I'll add it in this PR only

Reply:
added a check if it's not an empty list